doc/components/popup.md

```diff
@@ -0,0 +1,481 @@
+```
+
+###### Editor
+
+- Relative to the current editor on current window
+
+```lua
+relative = "editor"
+```
+
+###### Cursor (default)
+
+- Relative to the current window
+
+```lua
+relative = "win"
+```
+
+###### Target Window
+
+- Relative to the window of `id`
+
+```lua
+relative = {
+    type = "win",
+    winid = 5,
+}
+```
+
+###### Buffer position
+
+- Relative to the buffer's `position`
+
+```lua
+relative = {
+    type = "buf",
+    position = {
+        row = 5,
+        col = 5,
+    },
+}
+```
+
+---
+
+#### `position`
+
+- **Type:** `number` or `percentage string` or `table`
+- **Description:**
+  - If `position` is `number` or `percentage string`, it applies to both `row` and `col`
+  - Position is calculated from the top-left corner
+  - For `percentage`, position is calculated according to the option `relative`
+      - If `relative` is set to `"buf"` or `"cursor"`, `percentage string` is not allowed
+
+#### Usage
+
+```lua
+position = 50,
+```
+
+```lua
+position = "50%",
+```
+
+```lua
+position = {
+    row = 30,
+    col = 20
+}
+```
+
+```lua
+position = {
+    row = "30%",
+    col = "20%"
+}
+```
+
+---
+
+### `size`
+
+- **Type:** `number` or `percentage string` or `table`
+- **Description:**
+  - Determines the size of the popup
+  - For `percentage`, `size` is calculated according to the option `relative`
+    - If `relative` is set to `buf` or `cursor`, window `size` is considered
+  - If `size` is `number` or `percentage`, it applies to both `width` and `height`
+        - You can also pass a table to set them separately.
+
+#### Usage
+
+```lua
+size = 50,
+```
+
+```lua
+size = "50%",
+```
+
+```lua
+size = {
+    width = 30,
+    height = 50,
+}
+```
+
+```lua
+size = {
+    width = "30%",
+    height = "50%",
+}
+```
+
+---
+
+### `enter`
+
+- **Type:** `boolean`
+- **Description:** If `true`, the popup is auto-entered after mount
+
+#### Usage
+
+```lua
+enter = true,
+```
+
+---
+
+### `focusable`
+
+- **Type:** `boolean`
+- **Description:** Is `false`, the popup can not be entered by user actions (wincmds, mouse events)
+
+#### Usage
+
+```lua
+focusable = true,
+```
+
+---
+
+### `zindex`
+
+- **Type:** `number`
+- **Description:**
+  - Sets the order of the popup on z-axis
+  - Popup with higher the `zindex` goes on top of popups with lower `zindex`
+
+#### Usage
+
+```lua
+zindex = 20,
+```
+
+---
+
+### `buf_options`
+
+- **Type:** `table`
+- **Description:** Contains all buffer related options
+
+#### Usage
+
+```lua
+buf_options = {
+    modifiable = false,
+    readonly = true,
+},
+```
+
+#### Descriptions
+
+| Value        | Description                                                                  |
+| ------------ | ---------------------------------------------------------------------------- |
+| `modifiable` | If `false` the buffer contents cannot be changed. <br> Check `:h modifiable` |
+| `readonly`   | If `true`, writes fail unless you use a `!` <br> Check `:h readonly`         |
```

Comment:
Describing these here seems unnecessary, since `buf_options` can take other buffer optinos too (not just these two).

Reply:
Ohh, I didn't know it accepted other options. Could we point a user to a list of those options if they are in the help somewhere?

Reply:
Do `:help options` and search for `local to buffer`

Reply:
What could we write here so users can easily figure it out?

I went in options help and searched for local to buffer and it wasn't as straightforward as it could be.

Reply:
Well, I don't think there are any separate list for these options... they're all mixed together in `:help options`. So nothing much we can do here.

You can run `:help options | /local to buffer` and the search will be initiated for you. Then you'll just have to press `n` to go to next results.